lib/keystores.js

```diff
@@ -275,13 +271,8 @@ exports.getStorageUsage = async ({
     }
 
     // start counting keys in keystore
-    counters.push(_addKeyCount({usage, moduleApi, keystoreId, signal}));
-
-    // await counters if max concurrency reached
-    if(counters.length === USAGE_COUNTER_MAX_CONCURRENCY) {
-      await Promise.all(counters);
-      counters = [];
-    }
+    counters.push(limit(() => _addKeyCount({
+      usage, moduleApi, keystoreId, signal})));
 
     if(signal && signal.abort) {
```

Comment:
Hmm, this changes the effectiveness of this code. With `pLimit`, all calls that are deferred because of the limit will still execute after the abort signal has been triggered. Previously, only those calls that were already executing would be run and no further calls would be made if the abort happened while waiting on those. So, I think we need a separate helper that takes this signal:

```js
function _checkComputeStorageSignal({signal}) {
  // move above conditional and throw here
}
```

And then we need to call that both here -- and in the arrow functions that we pass to limit.